src/sort_filter/KalmanTracker.cpp

```diff
@@ -0,0 +1,117 @@
+///////////////////////////////////////////////////////////////////////////////
+// KalmanTracker.cpp: KalmanTracker Class Implementation Declaration
+
+#include "KalmanTracker.h"
+#include <ctime>
+
+using namespace std;
+using namespace cv;
+
+int KalmanTracker::kf_count = 0;
+
+// initialize Kalman filter
+void KalmanTracker::init_kf(
+	StateType stateMat)
+{
+	int stateNum = 7;
+	int measureNum = 4;
+	kf = KalmanFilter(stateNum, measureNum, 0);
+
+	measurement = Mat::zeros(measureNum, 1, CV_32F);
+
+	kf.transitionMatrix = (Mat_<float>(stateNum, stateNum) << 1, 0, 0, 0, 1, 0, 0,
+						   0, 1, 0, 0, 0, 1, 0,
+						   0, 0, 1, 0, 0, 0, 1,
+						   0, 0, 0, 1, 0, 0, 0,
+						   0, 0, 0, 0, 1, 0, 0,
+						   0, 0, 0, 0, 0, 1, 0,
+						   0, 0, 0, 0, 0, 0, 1);

```

Comment:
And, this is _why_ I personally hate tab indents. They only work if you always switch to spaces for **alignment** indentation.

Then again, I also hate blocks of code aligned way to the right side of the screen. IMHO this should be:
```suggestion
	kf.transitionMatrix = (Mat_<float>(stateNum, stateNum)
		<< 1, 0, 0, 0, 1, 0, 0,
		   0, 1, 0, 0, 0, 1, 0,
		   0, 0, 1, 0, 0, 0, 1,
		   0, 0, 0, 1, 0, 0, 0,
		   0, 0, 0, 0, 1, 0, 0,
		   0, 0, 0, 0, 0, 1, 0,
		   0, 0, 0, 0, 0, 0, 1);
```
And then it doesn't actually matter if you use spaces or tabs to indent.